### PR TITLE
Add error-prone CompileTimeConstantViolatesLiskovSubstitution check

### PR DESCRIPTION
**Pull request by carterkozak**
==COMMIT_MSG==
Add error-prone CompileTimeConstantViolatesLiskovSubstitution check
==COMMIT_MSG==

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -199,6 +199,7 @@ Safe Logging can be found at [github.com/palantir/safe-logging](https://github.c
 - `DeprecatedGuavaObjects`: `com.google.common.base.Objects` has been obviated by `java.util.Objects`.
 - `JavaTimeSystemDefaultTimeZone`: Avoid using the system default time zone.
 - `IncubatingMethod`: Prevents calling Conjure incubating APIs unless you explicitly opt-out of the check on a per-use or per-project basis.
+- `CompileTimeConstantViolatesLiskovSubstitution`: Requires consistent application of the `@CompileTimeConstant` annotation.
```
Comment (iamdanfox) [inline README.md]: I'd flesh this out to say something like "annotating a subclass method with `@CompileTimeConstant` but leaving a parent class/interface un-annotated may result in the validation being missed"

**Comment by iamdanfox** [comment]
This is pretty fun! How much do you think it will tank compiler performance?

**Comment by carterkozak** [comment]
I imagine it will be... measurable

File: README.md (hunk 1 above)
Comment (carterkozak) [reply README.md]: :+1:

**Review by iamdanfox — APPROVED** [review]

**Pull request merged**